### ISSUE TITLE
Issue with [project_files] section

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#789]
In GitLab by @dbeltrankyl on Mar 7, 2022, 16:57

Hello, 

I'm opening this issue due to a bug @gmontanepinto found related to the [project_files] section of the expdef.conf.

```
[project_files]
# Where is PROJECT CONFIGURATION file location relative to project root path
FILE_PROJECT_CONF = templates/conf/proj_GLOBAL_AEROSOLS.conf
# Where is JOBS CONFIGURATION file location relative to project root path
FILE_JOBS_CONF = templates/conf/jobs_GLOBAL_AEROSOLS.conf
# Default job scripts type in the project. type = STRING, default = bash, supported = 'bash', 'python' or 'r'
JOB_SCRIPTS_TYPE =
```

The bug is that if the files don't exist. Autosubmit will erase jobs.conf and proj.conf each time a refresh is prompt and then crash without any kind of information. 
* The delete of the jobs.conf/proj.conf is intended, but perhaps we should only rename it. And if the files are not found in the git repo, rename it back to jobs.conf or/and proj.conf.

This must be changed to make backups for the current configuration. (jobs.conf_N, proj.conf_n)
And implement an error catching mechanism to stop the execution in a sane way(or continue it but without coping  the proj/jobs conf). @mcastril

**Comment by kinow** [comment]
In GitLab by @mcastril on Mar 8, 2022, 10:55

> The bug is that if the files don't exist. Autosubmit will erase jobs.conf and proj.conf each time a refresh is prompt and then crash without any kind of information.

How can they be deleted if they do not exist? Do you mean defined?

> The delete of the jobs.conf/proj.conf is intended, but perhaps we should only rename it. And if the files are not found in the git repo, rename it back to jobs.conf or/and proj.conf.

What do you mean with rename it? Do you mean to do a backup of jobs.conf and projs.conf instead of removing them?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 8, 2022, 11:08

> How can they be deleted if they do not exist? Do you mean defined?

The experiment was a copy from another experiment with its own jobs.conf and proj.conf, and even if it is dummy, the jobs.conf is always present.

When [project_files] is set and an autosubmit refresh ( or create if proj doesn't exist) is prompt. The files ( in this case, jobs and proj) under $exp_root/conf will be erased. 

> What do you mean with rename it? Do you mean to do a backup of jobs.conf and proj.conf instead of removing them?

Yes, I mean to do a backup of the current configuration, instead of being erased and replaced by the [project_files]. So the change would be a rename + replace instead of erase + replace

**Comment by kinow** [comment]
In GitLab by @mcastril on Mar 8, 2022, 11:19

> The experiment was a copy from another experiment with its own jobs.conf and proj.conf, and even if it is dummy, the jobs.conf is always present.
> 
> When \[project_files\] is set and an autosubmit refresh ( or create if proj doesn't exist) is prompt. The files ( in this case, jobs and proj) under $exp_root/conf will be erased.

So you mean that they are removed in case that `FILE_PROJECT_CONF` and `FILE_JOBS_CONF` are not `defined` ? 

I think you didn't explain what did you mean with `The bug is that if the files don't exist.` (you say that they are always present).

**Comment by kinow** [comment]
In GitLab by @gmontanepinto on Mar 8, 2022, 11:35

The problem in my case was that the paths defined in `FILE_PROJECT_CONF` and `FILE_JOBS_CONF` were wrong and didn't exist. Then autosubmit just removed the existing jobs.conf and crashed because of the wrong paths without any error message.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 8, 2022, 11:54

> So you mean that they are removed in case that `FILE_PROJECT_CONF` and `FILE_JOBS_CONF` are not `defined` ?

No, they're removed in case these directives are defined. This is the feature of these parameters that replaces the current jobs.conf and PROJ*.conf for the ones that you define on there

> The bug is that if the files don't exist.

Sorry, as @gmontanepinto, said, I was referring to the case that the path you defined on there doesn't exist

**Comment by kinow** [comment]
In GitLab by @mcastril on Mar 8, 2022, 12:31

Ok, I got it now! Thank you!

I agree with the proposed solution.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 17, 2022, 10:34

The fix is now available on 3.14.0b
